src/LargeFileUploadTask.ts

```diff
@@ -0,0 +1,248 @@
+            }
+        };
+        try {
+            let session = await LargeFileUploadTask.createUploadSession(client, options.sessionRequestUrl, payload);
+            return new LargeFileUploadTask(client, _fileObject, session, options);
+        } catch(err) {
+            throw err;
+        }
+    }
+
+    // Makes call to server to create an uploadsession
+    static async createUploadSession(client: Client, requestUrl: string, requestPayload: any): Promise<any> {
+        try {
+            let createSession = await client.api(requestUrl).post(requestPayload);
+            return <LargeFileUploadSession>{
+                url: createSession.uploadUrl,
+                expiry: new Date(createSession.expirationDateTime)
+            };
+        } catch(err) {
+            throw err;
+        }
+    }
+
+    // Returns random filename created from the current time
+    static getRandomFileName(): string {
+        return `Upload_${new Date().toDateString().replace(/ /g, "_")}`;
+    }
+
+    // Parses the range given as the array of string and returns the first range as a Range Object
+    parseRange(ranges: string[]): Range {
+        let rangeStr = ranges[0];
+        if (typeof rangeStr === "undefined" || rangeStr === "") {
+            return new Range();
+        }
+        let firstRange = rangeStr.split("-"),
+            minVal = parseInt(firstRange[0]),
+            maxVal = parseInt(firstRange[1]);
+        if (Number.isNaN(maxVal)) {
+            maxVal = this.file.size - 1;
+        }
+        return new Range(minVal, maxVal);
+    }
+
+    // updates the task's exipration date and the next expected range
+    updateTaskStatus(response: UploadStatusResponse): void {
+        let self = this;
+        self.uploadSession.expiry = new Date(response.expirationDateTime);
+        self.nextRange = self.parseRange(response.nextExpectedRanges);
+    }
+
+    // Returns the next range that needs to be sent based on the rangeSize of the rask
+    getNextRange(): Range {
+        let self = this;
+        if (self.nextRange.minValue === -1) {
+            return self.nextRange;
+        }
+        let minVal = self.nextRange.minValue,
+            maxValue = minVal + self.options.rangeSize - 1;
+        if (maxValue >= self.file.size) {
+            maxValue = self.file.size - 1;
+        }
+        return new Range(minVal, maxValue);
+    }
+
+    // Slices the file to the given range offset
+    sliceFile(range: Range): ArrayBuffer | Blob {
+        let blob = this.file.content.slice(range.minValue, range.maxValue + 1);
+        return blob;
+    }
+
+    // Uploads the file by slicing and uploadin each slice in a sequential manner
+    async upload(): Promise<any> {
+        let self = this;
+        try {
+            while (true) {
+                let nextRange = self.getNextRange();
+                if (nextRange.maxValue === -1) {
+                    throw new Error("Invalid session: Uploading completed");
```

Comment:
This also aplies to line 181. Would it make sense to create the Error, and then give it a descriptive name so that customers can inspect the error.name and handle it? The error message should give some info on what happed advice on how the error can be handled.